test/realizations/catchments/Bmi_C_Adapter_Test.cpp

```diff
@@ -190,6 +193,71 @@ TEST_F(Bmi_C_Adapter_Test, Update_0_c) {
     adapter.Finalize();
 }
 
+/** Profile the update function and GetValues functions */
+TEST_F(Bmi_C_Adapter_Test, Profile)
+{
+    Bmi_C_Adapter adapter(lib_file_name_0, config_file_name_0, forcing_file_name_0, true, false, true,
+                          REGISTRATION_FUNC, utils::StreamHandler());
+
+    int num_ouput_items = adapter.GetOutputItemCount();
+    int num_input_items = adapter.GetInputItemCount();
+
+    std::vector<std::string> output_names = adapter.GetOutputVarNames();
+    std::vector<std::string> input_names = adapter.GetInputVarNames();
+
+    std::unordered_map<std::string, std::vector<long> > saved_times;
+
+
+    using time_point = std::chrono::time_point<std::chrono::steady_clock>;
+    auto to_micros = [](const time_point& s, const time_point& e){ return std::chrono::duration_cast<std::chrono::microseconds>(e - s).count();};
+
+    adapter.Initialize();
+    // Do the first few time steps
+    for (int i = 0; i < 720; i++)
+    {
+        // record time for Update
+        auto s1 = std::chrono::steady_clock::now();
+        adapter.Update();
+        auto e1 = std::chrono::steady_clock::now();
+        saved_times["Update"].push_back(to_micros(s1,e1));
+
+        // record time to get each output variable
+        for(std::string name : output_names)
+        {
+            auto s2 = std::chrono::steady_clock::now();
+            std::vector<double> values = adapter.GetValue<double>(name);
+            auto e2 = std::chrono::steady_clock::now();
+            saved_times["Get " + name].push_back(to_micros(s2,e2));
+        }
+
+        // record time to get each input variable
+        for(std::string name : input_names)
+        {
+            auto s3 = std::chrono::steady_clock::now();
+            std::vector<double> values = adapter.GetValue<double>(name);
+            auto e3 = std::chrono::steady_clock::now();
+            saved_times["Get " + name].push_back(to_micros(s3,e3));
+        }
+
+    }
+
+    // calcuate average time for each record
+
+    for ( auto& kv : saved_times )
+    {
+        auto key = kv.first;
+        auto& v = kv.second;
+        long sum = 0;
+
+        sum = std::accumulate(v.begin(), v.end(), sum );
+        double average = double(sum) / v.size();
+
+        std::cout << "Average time for " << key << " = " << average << "µs\n";
+    }
+
+    adapter.Finalize();
+}
+
```

Comment:
Did you intend to add this test for the BMI C adapter in this change set?

Reply:
I did not change this file. It might have come from somewhere else or merging from upstream master branch.

Reply:
If you copied something that Donald sent you for the partitioning work, I wonder if it that's where it came from.  He was working on the profiling earlier.  An unintentional addition along those lines would make sense.  

Regardless, I suggest taking these out of this PR.  If you are doing a rebase anyway for the CMake changes I mentioned elsewhere, you could rebase interactively and take these changes out during the rebase.

Reply:
I dont think I touched that file

Reply:
I  did a rebase and re-testing before reading the new comments. Will work on it.